syntax="proto3";

package fixbug;

option cc_generic_services=true;

message ResultCode{
    int32 errcode=1;
    bytes errmsg=2;
}

// 参数类型
message LoginRequest{
    bytes name=1;
    bytes pwd=2;
}

// 返回值类型
message LoginReponse{
    ResultCode result=1;
    bool success=2;
}

message RegisterRequest{
    uint32 id=1;
    bytes name=2;
    bytes pwd=3;

}

message RegisterResponse{
    ResultCode result=1;
    bool success=2;
}

service UserServiceRpc{
    // 这里是函数描述
    rpc Login(LoginRequest) returns(LoginReponse);
    rpc Register(RegisterRequest) returns(RegisterResponse);
}